File: Pandacap.Data/DeviantArtCredentials.cs
using Microsoft.AspNetCore.Identity;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Pandacap.Data
{
    public class DeviantArtCredentials
    {
        [Key]
        public string UserId { get; set; } = "";

        [ForeignKey(nameof(UserId))]
        public IdentityUser? User { get; set; }

        [Required]
        public string AccessToken { get; set; } = "";

        [Required]
        public string RefreshToken { get; set; } = "";
    }
}

File: Pandacap.Data/DeviantArtCredentials.fs
namespace Pandacap.Data

open System.ComponentModel.DataAnnotations

/// The active credentials for the DeviantArt API.
type DeviantArtCredentials() =
    [<Key>]
    member val Username = "" with get, set

    member val AccessToken = "" with get, set
    member val RefreshToken = "" with get, set
